enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/NeoFullRESTInteraction.java

```diff
@@ -0,0 +1,215 @@
+                .build();
+        server.start();
+    }
+
+    @Override
+    public EnterpriseUserManager getManager()
+    {
+        return server.getDependencyResolver().resolveDependency( EnterpriseAuthManager.class ).getUserManager();
+    }
+
+    @Override
+    public String executeQuery( RESTSubject subject, String call, Map<String,Object> params,
+            Consumer<ResourceIterator<Map<String, Object>>> resultConsumer )
+    {
+        String escapedCall = call.replace( "'", "\\'" ).replace( "\"", "\\\"" );
+        HTTP.RawPayload payload = HTTP.RawPayload.quotedJson(
+                "{'statements':[{'statement':'" + escapedCall + "'}]}" );
+        HTTP.Response response = HTTP.withHeaders( HttpHeaders.AUTHORIZATION, subject.principalCredentials )
+                .request( POST, commitURL(), payload );
+
+        try
+        {
+            JsonNode data = JsonHelper.jsonNode( response.rawContent() );
+            if ( data.has( "errors" ) && data.get( "errors").has( 0 ) )
+            {
+                JsonNode firstError = data.get( "errors" ).get( 0 );
+                if ( firstError.has( "message" ) )
+                {
+                    return firstError.get( "message" ).asText();
+                }
+            }
+            if ( data.has( "results" ) && data.get( "results" ).has( 0 ) )
+            {
+                JsonNode firstResult = data.get( "results" ).get( 0 );
+                RESTResult result = new RESTResult( firstResult );
+                resultConsumer.accept( result );
+            }
+        }
+        catch ( JsonParseException e )
+        {
+            fail("Unexpected error parsing Json!");
+        }
+
+        return "";
+    }
+
+    @Override
+    public RESTSubject login( String username, String password ) throws Throwable
+    {
+        String principalCredentials = challengeResponse( username, password );
+        HTTP.Response response = authenticate( principalCredentials );
+        return new RESTSubject( response, principalCredentials );
+    }
+
+    private HTTP.Response authenticate( String principalCredentials )
+    {
+        return HTTP.withHeaders( HttpHeaders.AUTHORIZATION, principalCredentials ).request( POST, commitURL() );
+    }
+
+    @Override
+    public void logout( RESTSubject subject ) { }
+
+    @Override
+    public boolean isAuthenticated( RESTSubject subject )
+    {
+        subject.response = authenticate( subject.principalCredentials );
+        return subject.response.status() == 200 || // OK
+                subject.response.status() == 403; // PASSWORD CHANGE REQUIRED
+    }
+
+    @Override
+    public AuthenticationResult authenticationResult( RESTSubject subject )
+    {
+        switch ( subject.response.status() )
+        {
+            case 200: return AuthenticationResult.SUCCESS;
+            case 403: return AuthenticationResult.PASSWORD_CHANGE_REQUIRED;
```

Comment:
The response should also contain the element "password_change". 403 is also used for AuthorizationViolationException "Unauthorized access violation: %s."